resources/check.rb

```diff
@@ -19,12 +19,22 @@
 attribute :timeout, :kind_of => Integer
 attribute :ttl, :kind_of => Integer
 attribute :type, :kind_of => String, :equal_to => %w[standard status metric]
+attribute :cron, :kind_of => String
 
 def after_created
   unless name =~ /^[\w\.-]+$/
     raise Chef::Exceptions::ValidationFailed, "Sensu check #{name}: name cannot contain spaces or special characters"
   end
 
+  if !cron.nil? && !interval.nil?
+    raise Chef::Exceptions::ValidationFailed, "Sensu check #{name}: must either define interval, or cron, not both."
+  end
+
+  # preserve existing defalut value for interval
+  if cron.nil? && interval.nil?
```

Comment:
I see this is where you are preserving backwards compatibility, which I thank you for doing so. I think there is a more straightforward way than this as I have outlined above.